wallet/src/main/java/com/android/identity_credential/wallet/PresentationActivity.kt

```diff
@@ -217,159 +146,181 @@ class PresentationActivity : FragmentActivity() {
-                    consentData = consentPromptData.value
-                    if (consentData != null) {
-                        ConsentPromptEntryField(
-                            consentData = consentData!!,
-                            documentTypeRepository = walletApp.documentTypeRepository,
-                            onConfirm = { // user accepted to send requested credential data
-                                finishProcessingRequest()
-                            },
-                            onCancel = { // user declined submitting data to requesting party
-                                finish() // close activity
-                            }
-                        )
-                    }
+                State.RESPONSE_SENT -> {
+                    Logger.i(TAG, "State: Response Sent")
+                    // cleanup
+                    disconnect()
                 }
             }
         }
     }
 
-    /**
-     * Set property transferHelper to a new instance of TransferHelper created with a new instance
-     * of DeviceRetrievalHelper
-     */
-    private fun newTransferHelper(deviceRetrievalHelper: DeviceRetrievalHelper) {
-        // new TransferHelper instance
-        transferHelper =
-            transferHelperBuilder.setDeviceRetrievalHelper(deviceRetrievalHelper).build()
-    }
 
     /**
-     * Create a new instance of DeviceRetrievalHelper and,
-     * 1. set property deviceRetrievalHelper to point to it
-     * 2. return the new instance so it can be used to create a new TransferHelper
+     * Find a suitable Document for the given [docRequest] else throw [IllegalStateException].
+     * @param docRequest parsed from DeviceRequest CBOR.
+     * @return a matching [Document] from either on-screen Document or [DocumentStore].
+     * @throws IllegalStateException if no Documents in [DocumentStore] matched the given [docType]
      */
-    private fun newDeviceRetrievalHelper() =
-        DeviceRetrievalHelper.Builder(
-            applicationContext,
-            object : DeviceRetrievalHelper.Listener {
-
-                override fun onEReaderKeyReceived(eReaderKey: EcPublicKey) {
-                    Logger.i(TAG, "onEReaderKeyReceived")
-                }
-
-                override fun onDeviceRequest(deviceRequestBytes: ByteArray) {
-                    Logger.i(TAG, "onDeviceRequest")
-                    deviceRequest = deviceRequestBytes
-                    state.value = State.REQUEST_AVAILABLE
-                }
-
-                override fun onDeviceDisconnected(transportSpecificTermination: Boolean) {
-                    Logger.i(TAG, "onDeviceDisconnected $transportSpecificTermination")
-                    deviceRetrievalHelper?.disconnect()
-                    deviceRetrievalHelper = null
-                    state.value = State.NOT_CONNECTED
+    fun findSuitableDocumentForRequest(docRequest: DeviceRequestParser.DocRequest): Document {
```

Comment:
As discussed should be called `findMdocCredentialForRequest()` and return a `MdocCredential?`